### PR TITLE
[ENG-436] Fix preprint metrics

### PR DESCRIPTION
**Pull request by brianjgeiger**
## Purpose

Preprint metrics were initially counting days rather than tallying total downloads/views. This corrects that.

## Changes

1. Make it so future impact indices will include their source info (including if we re-export or gather node-file/quick-file metrics).
2. Fix the query and output so it includes all of the proper information.

## QA Notes

No QA

## Documentation

This shouldn't require any documentation changes.

## Side Effects

No.

## Ticket

https://openscience.atlassian.net/browse/ENG-436

### CODE REVIEW

File: osf/metrics.py [hunk 1]
```diff
@@ -135,6 +135,7 @@ class Index:
 
     class Meta:
         abstract = True
+        source = metrics.MetaField(enabled=True)
```
Comment (UdayVarkhedkar) [inline osf/metrics.py]: Why is the `_source` field necessary?

**Comment by UdayVarkhedkar** [comment]
Looks good to me!

I acknowledged that the `_source` field was enabled for future indices which is why I didn't initially understand the functional need for it here.

For others/future reference: I found it helpful to return `JsonResponse(response.to_dict())` on line 97 of api/metrics/views.py to see the pre-formatted results of the elasticsearch query and better understand the implications of the changes made.

**Comment by brianjgeiger** [comment]
The main reasons for the source field are related to finding out what's happening on production. Also, you can always strip the source field, but you can never put it back without regenerating the source, and that's not always possible.

File: api/metrics/views.py [hunk 2]
```diff
@@ -90,8 +89,9 @@ def get(self, *args, **kwargs):
 
         search = self.metric.search(after=start_datetime)
         search = search.filter('range', timestamp={'gte': start_datetime, 'lt': end_datetime})
-        search.aggs.bucket('preprints_per_day', 'date_histogram', field='timestamp', interval=interval)
-        search.aggs['preprints_per_day'].metric('per_preprint', 'terms', field='preprint_id')
+        search.aggs.bucket('dates', 'date_histogram', field='timestamp', interval=interval) \
+            .bucket('preprints', 'terms', field='preprint_id') \
+            .metric('total', 'sum', field='count')
```
Comment (pattisdr) [inline api/metrics/views.py]: 💯

**Comment by pattisdr** [comment]
Going ahead and merging straight to develop - no QA required

**Pull request merged**